dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/enrollment/DefaultEnrollmentService.java

```diff
@@ -92,16 +94,21 @@ public Enrollment getEnrollment(String uid, EnrollmentParams params, boolean inc
       throw new NotFoundException(Enrollment.class, uid);
     }
 
-    return getEnrollment(enrollment, params, includeDeleted);
+    return getEnrollment(enrollment, params, includeDeleted, null);
   }
 
   @Override
   public Enrollment getEnrollment(
-      @Nonnull Enrollment enrollment, EnrollmentParams params, boolean includeDeleted)
+      @Nonnull Enrollment enrollment,
+      EnrollmentParams params,
+      boolean includeDeleted,
+      OrganisationUnitSelectionMode orgUnitMode)
       throws ForbiddenException {
     User user = currentUserService.getCurrentUser();
     List<String> errors = trackerAccessManager.canRead(user, enrollment, false);
-    if (!errors.isEmpty()) {
+    boolean skipValidation =
+        user.isAuthorized(F_TRACKED_ENTITY_INSTANCE_SEARCH_IN_ALL_ORGUNITS) && orgUnitMode == ALL;
```

Comment:
Based on operation params validations, could we assume that if the mode is ALL, then it was already verified that the user has such authority?
Maybe it is not a big difference in the code but the concept is that after the operation validation, you can trust the orgUnitMode that is there

Reply:
In this case we can't, because it could be the user has the authority `ALL`, meaning it's a superuser.
But this just gave me a better idea on how to handle that, I'll refactor this a bit more.